services/jobs/Dockerfile

```diff
@@ -1,35 +1,72 @@
 FROM python:3.8
 
+# Install system dependencies
 RUN apt-get -qq update && \
     apt-get -q -y upgrade && \
-    apt-get install -y sudo curl wget locales && \
+    apt-get install -y \
+        sudo \
+        curl \
+        wget \
+        locales \
+        parallel \
+        time \
+        default-jre \
+        default-jdk \
+        python3-distutils \
+        openjdk-17-jre \
+        openjdk-17-jdk \
+        raptor2-utils \
+        jq && \
+    locale-gen en_US.UTF-8 && \
     rm -rf /var/lib/apt/lists/*
-RUN locale-gen en_US.UTF-8
 
-RUN apt clean
-RUN apt update
-RUN apt install -y default-jre jq raptor2-utils
+ENV LANG=en_US.UTF-8 \
+    LANGUAGE=en_US:en \
+    LC_ALL=en_US.UTF-8
+
 # Install Python packages
-RUN pip install chardet date-parser-sari lxml urllib3 requests edtf tqdm rdflib sari-field-definitions-generator sparqlwrapper PyYAML Pillow beautifulsoup4 Sickle dicttoxml Pillow pytz pyshacl pandas semantic-field-definition-generator pyexistdb pyduktape2
+RUN pip install --no-cache-dir \
+    chardet \
+    date-parser-sari \
+    lxml \
+    urllib3 \
+    requests \
+    edtf \
+    tqdm \
+    rdflib \
+    sari-field-definitions-generator \
+    sparqlwrapper \
+    PyYAML \
+    Pillow \
+    pytz \
+    pyshacl \
+    pandas \
+    semantic-field-definition-generator \
+    pyexistdb \
+    pyduktape2 \
+    python-decouple \
+    requests-toolbelt
 
 # Install x3ml Mapping engine
-RUN mkdir x3ml
-RUN wget https://github.com/isl/x3ml/releases/download/2.2.0/x3ml-engine-2.2.0-exejar.jar
-RUN mv x3ml-engine-2.2.0-exejar.jar x3ml/x3ml-engine.exejar
+RUN mkdir -p /libs /java/bin /x3ml /scripts /data /mapping
+RUN wget https://github.com/isl/x3ml/releases/download/2.2.0/x3ml-engine-2.2.0-exejar.jar -O /libs/x3ml-engine-exejar.jar
+
+# COPY x3ml service
+COPY x3ml/java/bin/ /java/bin/
+
+# Install task runner (http://taskfile.dev)
+RUN sh -c "$(curl -sSL https://taskfile.dev/install.sh)" -- -d
 
 # Install AWS CLI
 RUN pip --no-cache-dir install --upgrade awscli
 
-# Install task runner (http://taskfile.dev)
-RUN sh -c "$(curl -ssL https://taskfile.dev/install.sh)" -- -d
+# Set Java options
+ENV JAVA_OPTS="-Xms512m -Xmx1024m"
 
-# Prepare directories and volumes
-RUN mkdir /data
-RUN mkdir /mapping
-RUN mkdir /scripts
+# Define volumes and working directory
 VOLUME /data
 VOLUME /mapping
 WORKDIR /scripts
 
-# Run idling
-ENTRYPOINT tail -f /dev/null
\ No newline at end of file
+# Entry point
+ENTRYPOINT ["java", "-Xms512m", "-Xmx1024m", "-cp", "/libs/x3ml-engine-exejar.jar:/java/bin", "X3MLEngineService"]
```

Comment:
are the java opts necessary here?

Reply:
Not strictly necessary; removed them

Reply:
We need the x3ml-engine-exejar.jar because the service is using classes from it